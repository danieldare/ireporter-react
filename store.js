import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import rootReducer from './src/reducers';

const initialState = {};

const middlewares = [thunk];

if (process.env.NODE_ENV === `development`) {
  middlewares.push(createLogger());
}

const store = createStore(
  rootReducer,
  initialState,
  composeWithDevTools(applyMiddleware(...middlewares))
);

export default store;
