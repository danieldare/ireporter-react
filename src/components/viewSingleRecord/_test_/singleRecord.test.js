import React from 'react';
import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import ViewSingleRecord from '../viewSingleRecord';
import store from '../../../../store';
import mockRouterOptions from '../../../../test/__mocks__/mockRouter';

describe('<ViewSingleRecord/>', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(
      <Provider store={store}>
        <Router>
          <ViewSingleRecord />
        </Router>
      </Provider>,
      mockRouterOptions
    );
  });

  it('should match snapshot', () => {
    const tree = toJSON(wrapper);
    expect(tree).toMatchSnapshot();
  });
});
