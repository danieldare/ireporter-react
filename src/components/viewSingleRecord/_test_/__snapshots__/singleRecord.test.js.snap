// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleIncident /> rendering should render without crashing 1`] = `ReactWrapper {}`;
